elfpy/utils/apeworx_integrations.py

```diff
@@ -0,0 +1,250 @@
+"""Helper functions for integrating the sim repo with solidity contracts via Apeworx"""
+
+from __future__ import annotations
+from typing import TYPE_CHECKING, Any
+from enum import IntEnum
+
+import logging
+
+# TODO: apeworx is not worxing with github actions when it is listed in requirements
+# and pyright doesn't like imports that aren't also in requirements.
+# pylint: disable=import-error
+import ape  # type: ignore[reportMissingImports]
+
+if TYPE_CHECKING:
+    from ape.types import ContractLog  # type: ignore[reportMissingImports]
+    from ape.api.accounts import AccountAPI  # type: ignore[reportMissingImports]
+    from ape_ethereum.transactions import Receipt  # type: ignore[reportMissingImports]
+    from ape.contracts.base import ContractInstance  # type: ignore[reportMissingImports]
+
+
+class AssetIdPrefix(IntEnum):
+    r"""The asset ID is used to encode the trade type in a transaction receipt"""
+    LONG = 0
+    SHORT = 1
+    WITHDRAWAL_SHARE = 2
+
+
+def encode_asset_id(prefix: int, timestamp: int) -> int:
```

Comment:
nice! but `encode_asset_id` and `decode_asset_id` definitely don't belong in apeworx_integrations.py.  definitely more of a hyperdrive_market.py thing.  They are very specific to the implementation of the hyperdrive smart contract